presto-hive/src/main/java/com/facebook/presto/hive/HiveHdfsConfiguration.java

```diff
@@ -25,7 +25,20 @@
 public class HiveHdfsConfiguration
         implements HdfsConfiguration
 {
-    private static final Configuration DEFAULT_CONFIGURATION = new Configuration();
+    private static final Configuration INITIAL_CONFIGURATION;
+
+    static {
+        Configuration.addDefaultResource("hdfs-default.xml");
```

Comment:
What i'm trying to achive here is to make INITIAL_CONFIGURATION immutable. 
Before this patch default values from `hdfs-default.xml` were lazy-loaded during the DistributedFileSystem initialization. We are building cached configuration based on the INITIAL_CONFIGURATION, so, if the default values are loaded implicitly and in lazy manner we would have configurations in configuration cache. 
Loading default values from `hdfs-default` for each thread in ThreadLocal#initValue is also an option. But i thought that instead of reading the same XML file many times it is better to read it once, so we don't waste CPU cycles for every new Thread.